### PR TITLE
Enforce validation on select fields on employer edit page

### PR DESCRIPTION
**Pull request by cfeeney-va**
Addresses two requested validations:
1) Hiring commitment should be greater than or equal to one
2) EIN should be 9 characters long (originally requested in #131 )

Final validation on EIN (no duplicates) will be addressed in separate task.

Let me know if you have any questions!

Discussed design somewhat with Gina but we could still run the final product by her...ping me if you want me to add screenshots.

### CODE REVIEW

**Comment by ayaleloehr** [comment]
@cfeeney-va Sometimes entering a 0 commitment gives an error saying `Commit to hire must be a postive number` and sometimes it says `Commit to hire must be greater than or equal to 1`.  Can we standardize these?  And also I think it might be more accurate to say `Commitment must either be blank or a positive number`?

**Comment by cfeeney-va** [comment]
Interesting...I guess it depends on the intent of requiring non-zero commitments? If we're not requiring companies to make a commitment, I'm actually not sure why they can't enter 0; do you have more background on this?  I would think that leaving the error message the way it is would encourage making a commitment, but they one you suggested is more technically true...

File: app/assets/javascripts/employers_edit.js.erb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+function fields_have_errors() {
+  var hiring_commitment = $("#employer_commit_to_hire").val();
+  var ein = $("#employer_ein").val();
+  var num_text = '';
+  var error_text = '';
+  if (hiring_commitment && (hiring_commitment < 1)) {
+    num_text = '1 error';
+    error_text = '<li>Commit to hire must be a postive number</li>';
+  }
+  if (ein && (ein.length < 9)) {
```
Comment (ayaleloehr) [inline app/assets/javascripts/employers_edit.js.erb]: This `<` allows EIN's of length 10 to not throw an error. Should it be `!=` instead?

File: app/views/employers/_form.html.erb [hunk 2]
```diff
@@ -167,25 +165,30 @@
...
-		  <div class="small-12 columns">
+    <div class="row">
+      <div class="small-12 columns">
         <div class="actions">
-    		  <%= f.submit class: "button primary" %>
-		    </div>
+	  <% if current_user.va_admin? %>
+	    <%= f.submit class: "button primary", id: "submit-button" %>
+	  <% else %>
+	    <span id="click-button" class="button primary">Update Employer</span>
```
Comment (ayaleloehr) [inline app/views/employers/_form.html.erb]: I'm still struggling to explain why there are different buttons for admins vs regular users?  Why do we want different validations for these two categories?

File: app/assets/javascripts/employers_edit.js.erb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+function fields_have_errors() {
+  var hiring_commitment = $("#employer_commit_to_hire").val();
+  var ein = $("#employer_ein").val();
+  var num_text = '';
+  var error_text = '';
+  if (hiring_commitment && (hiring_commitment < 1)) {
+    num_text = '1 error';
+    error_text = '<li>Commit to hire must be a postive number</li>';
```
Comment (ayaleloehr) [inline app/assets/javascripts/employers_edit.js.erb]: change `postive` to `positive`

**Comment by ayaleloehr** [comment]
I don't have anymore background on the above about commitments >0, just trying to come up with the best user experience! I'm fine with it the way it is now actually; will encourage commitments!  

@cfeeney-va A few other comments above, all small. Let me know when you get to those and then I'll merge!

**Comment by cfeeney-va** [comment]
I had two buttons because I was thinking admins would only be adding notes, etc. to the profiles and we wouldn't need to run the validation. However, they could be changing these values so I think you're right that there should only be one configuration.  When I changed it, though a test keeps failing and I can't figure it out.  The line that's not working is: "fill_in "employer_admin_notes", with: 'Called and left message'."  The test is no longer filling in that text field which doesn't make sense to me...do you have any thoughts?

**Comment by ayaleloehr** [comment]
The issue was that the EIN of the employer in that test wasn't 9 digits, so the admin note was not being saved and the employer was not updating, thus no success "flash_notice" was at the top of the page.  This PR LGTM now, but since I pushed that last code change will let @cfeeney-va look then merge.

**Comment by cfeeney-va** [comment]
LGTM!  Thanks so much for tracking that down!

**Pull request merged**